File: VirtualWorld-Cpp/Virtual-World/Organizm.cpp
#include "Organizm.h"


int Organizm::getInicjatywa()
{
	return inicjatywa;
}

void Organizm::changeInicjatywa(int i)
{
	inicjatywa = i;
}

void Organizm::changeWiek(int i)
{
	wiek = i;
}

int Organizm::getWiek()
{
	return wiek;
}

int Organizm::getPolozenie(int i)
{
	return polozenie[i];
}

std::vector<int>& Organizm::getPolozenie()
{
	return polozenie;
}

void Organizm::changePolozenie(int x, int y)
{
	polozenie[0] = x;
	polozenie[1] = y;
}

Swiat* Organizm::getSwiat()
{
	return swiat;
}

void Organizm::changeSwiat(Swiat* swiat)
{
	this->swiat = swiat;

}

Organizm::Organizm()
{
	this->polozenie = std::vector<int>{ 0,0 };
}

Organizm::~Organizm()
{
}

void Organizm::akcja(std::vector<int>& blackList, std::vector<Organizm*>& sortedList, std::vector <std::vector <Organizm*> >& organizmy)
{
}

void Organizm::kolizja(std::vector<int> teoretyczna, std::vector<int>& blackList, std::vector<Organizm*>& sortedList, Organizm* uderzony, Organizm* uderzajacy, std::vector <std::vector <Organizm*>>& organizmy)
{
}

void Organizm::rysowanie()
{
}

char Organizm::getZnak()
{
	return 0;
}

bool Organizm::czyOdbito(Organizm* atakujacy)
{
	return false;
}

bool Organizm::zmienPolozenie(Swiat* swiat)
{
	return true;
}

bool Organizm::dobryWech(Organizm* atakowany)
{
	return false;
}

int Organizm::zasieg()
{
	return 1;
}

bool Organizm::isPlant()
{
	return false;
}

int Organizm::iloscRozp()
{
	return 1;
}

bool Organizm::ucieczka(std::vector<int>& teoretyczna, std::vector <std::vector <Organizm*>>& organizmy)
{
	return 0;
}

void Organizm::odstraszenieZPola(std::vector<int>& poleAtakujacego, std::vector<int>& poleAtakowanego, std::vector<std::vector<Organizm*>>& organizmy, char odstraszonego)
{
}

int Organizm::dodawanaSilaPoZjedzeniu()
{
	return 0;
}

bool Organizm::czyZabojcza()
{
	return false;
}

bool Organizm::isRadioactive()
{
	return false;
}

int Organizm::getKierunekRuchu()
{
	return 0;
}

void Organizm::zmienKierunekRuchu(int a)
{
}

void Organizm::podajKierunekRuchu()
{
}

void Organizm::podajInfoTarcza(int tura)
{
}

bool Organizm::czyJestTarczaAldura()
{
	return false;
}

void Organizm::zmienStanTarczyAldura(bool b)
{
}

int Organizm::getSila()
{
	return sila;
}

void Organizm::zmienSile(int s)
{
	sila = s;
}

File: VirtualWorld-Java/src/Organizm.java
import java.util.ArrayList;

public abstract class Organizm {

private Swiat swiat;
private ArrayList<Integer> polozenie;
private int sila;
private int inicjatywa;
private int wiek;

public int getInicjatywa() {
	return inicjatywa;
}

public void changeInicjatywa(int i) {
	this.inicjatywa = i;
}

public void changeWiek(int i) {
	wiek = i;
}

public int getWiek() {
	return wiek;
}

public void addOneToWiek() {
	wiek++;
}

public int getPolozenie(int i) {
	return polozenie.get(i);
}

public ArrayList<Integer> getPolozenie(){
	return polozenie;
}

public void changePolozenie(int x, int y) {
	polozenie.set(0, x);
	polozenie.set(1, y);
}

public Swiat getSwiat() {
	return swiat;
}

public void changeSwiat(Swiat swiat) {
	this.swiat = swiat;
}

public Organizm(){
	this.polozenie = new ArrayList<Integer>();
	polozenie.add(0);
	polozenie.add(0);
}

public char getZnak() {
	return 0;
}

public Boolean czyOdbito(Organizm atakujacy) {
	return false;
}

public Boolean zmienPolozenie(Swiat swiat) {
	return true;
}

public Boolean dobryWech(Organizm atakowany) {
	return false;
}

public int zasieg() {
	return 1;
}

public Boolean isPlant() {
	return false;
}

public int iloscRozp() {
	return 1;
}

public Boolean ucieczka(ArrayList<Integer> teoretyczna, ArrayList<ArrayList<Organizm>> organizmy) {
	return false;
}

public void odstraszenieZPola(ArrayList<Integer> poleAtakujacego, ArrayList<Integer> poleAtakowanego, ArrayList<ArrayList<Organizm>> organizmy, char odstraszonego) {
	
}

public int dodawanaSilaPoZjedzeniu() {
	return 0;
}

public Boolean czyZabojcza() {
	return false;
}

public Boolean isRadioactive() {
	return false;
}

public int getKierunekRuchu() {
	return 0;
}

public void zmienKierunekRuchu(int a) {
}

public String podajKierunekRuchu() {
	return null;
}

public String podajInfoTarcza(int tura) {
	return null;
}

public Boolean czyJestTarczaAldura() {
	return false;
}

public void zmienStanTarczyAldura(Boolean b) {
}

public int getSila() {
	return sila;
}

public void zmienSile(int s) {
	sila = s;
}

public void rysowanie() {	
}


public void akcja(ArrayList<Integer> blackList, ArrayList<Organizm> sortedList,
		ArrayList<ArrayList<Organizm>> organizmy) {

	
}

public void kolizja(ArrayList<Integer> teoretyczna, ArrayList<Integer> blackList, ArrayList<Organizm> sortedList,
		Organizm uderzony, Organizm uderzajacy, ArrayList<ArrayList<Organizm>> organizmy) {
	
}

	
}
